src/cpp/src/image_generation/flux_pipeline.hpp

```diff
@@ -0,0 +1,382 @@
+        }
+
+        // initialize generation config
+        initialize_generation_config(data["_class_name"].get<std::string>());
+    }
+
+    FluxPipeline(PipelineType pipeline_type,
+                 const std::filesystem::path& root_dir,
+                 const std::string& device,
+                 const ov::AnyMap& properties)
+        : DiffusionPipeline(pipeline_type) {
+        const std::filesystem::path model_index_path = root_dir / "model_index.json";
+        std::ifstream file(model_index_path);
+        OPENVINO_ASSERT(file.is_open(), "Failed to open ", model_index_path);
+
+        nlohmann::json data = nlohmann::json::parse(file);
+        using utils::read_json_param;
+
+        set_scheduler(Scheduler::from_config(root_dir / "scheduler/scheduler_config.json"));
+
+        const std::string text_encoder = data["text_encoder"][1].get<std::string>();
+        if (text_encoder == "CLIPTextModel") {
+            m_clip_text_encoder = std::make_shared<CLIPTextModel>(root_dir / "text_encoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", text_encoder, "' text encoder type");
+        }
+
+        const std::string t5_text_encoder = data["text_encoder_2"][1].get<std::string>();
+        if (t5_text_encoder == "T5EncoderModel") {
+            m_t5_text_encoder = std::make_shared<T5EncoderModel>(root_dir / "text_encoder_2", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", t5_text_encoder, "' text encoder type");
+        }
+
+        const std::string vae = data["vae"][1].get<std::string>();
+        if (vae == "AutoencoderKL") {
+            m_vae = std::make_shared<AutoencoderKL>(root_dir / "vae_decoder", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", vae, "' VAE decoder type");
+        }
+
+        const std::string transformer = data["transformer"][1].get<std::string>();
+        if (transformer == "FluxTransformer2DModel") {
+            m_transformer = std::make_shared<FluxTransformer2DModel>(root_dir / "transformer", device, properties);
+        } else {
+            OPENVINO_THROW("Unsupported '", transformer, "'Transformer type");
+        }
+
+        // initialize generation config
+        initialize_generation_config(data["_class_name"].get<std::string>());
+    }
+
+    FluxPipeline(PipelineType pipeline_type,
+                 const CLIPTextModel& clip_text_model,
+                 const T5EncoderModel& t5_text_model,
+                 const FluxTransformer2DModel& transformer,
+                 const AutoencoderKL& vae)
+        : DiffusionPipeline(pipeline_type),
+          m_clip_text_encoder(std::make_shared<CLIPTextModel>(clip_text_model)),
+          m_t5_text_encoder(std::make_shared<T5EncoderModel>(t5_text_model)),
+          m_vae(std::make_shared<AutoencoderKL>(vae)),
+          m_transformer(std::make_shared<FluxTransformer2DModel>(transformer)) {}
```

Comment:
let's initialize generation config here similarly to https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/image_generation/stable_diffusion_3_pipeline.hpp#L205